drivers/accel/adxl367/adxl367.c

```diff
@@ -0,0 +1,1310 @@
+ * @brief Writes data into a register.
+ *
+ * @param dev              - The device structure.
+ * @param register_value   - Data value to write.
+ * @param register_address - Address of the register.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_set_register_value(struct adxl367_dev *dev,
+				   uint8_t register_value,
+				   uint8_t register_address)
+{
+	uint8_t buffer[3] = {0, 0, 0};
+
+	if (dev->comm_type == ADXL367_SPI_COMM) {
+		buffer[0] = ADXL367_WRITE_REG;
+		buffer[1] = register_address;
+		buffer[2] = register_value;
+		return no_os_spi_write_and_read(dev->spi_desc,
+						buffer, 3);
+	} else {
+		buffer[0] = dev->i2c_slave_address + ADXL367_I2C_WRITE;
+		buffer[1] = register_address;
+		buffer[2] = register_value;
+		return no_os_i2c_write(dev->i2c_desc, buffer, 3, 1);
+	}
+
+}
+
+/***************************************************************************//**
+ * @brief Performs a burst read of a specified number of registers.
+ *
+ * @param dev              - The device structure.
+ * @param read_data        - The read values are stored in this buffer.
+ * @param register_address - The start address of the burst read.
+ * @param register_nb	   - Number of registers to be read in burst.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_get_register_value(struct adxl367_dev *dev,
+				   uint8_t* read_data,
+				   uint8_t  register_address,
+				   uint8_t  register_nb)
+{
+	int32_t ret;
+	uint8_t buffer[0x45];
+	uint8_t index;
+
+	memset(buffer, 0, 0x45);
+
+	if (dev->comm_type == ADXL367_SPI_COMM) {
+		buffer[0] = ADXL367_READ_REG;
+		buffer[1] = register_address;
+		ret = no_os_spi_write_and_read(dev->spi_desc, buffer, register_nb + 2);
+		if (ret)
+			return ret;
+		for (index = 0; index < register_nb; index++)
+			read_data[index] = buffer[index + 2];
+	} else {
+		buffer[0] = dev->i2c_slave_address + ADXL367_I2C_READ;
+		buffer[1] = register_address;
+		ret = no_os_i2c_write(dev->i2c_desc, buffer, 2, 0);
+		if (ret)
+			return ret;
+		ret = no_os_i2c_read(dev->i2c_desc, buffer, register_nb + 1, 1);
+		if (ret)
+			return ret;
+		for (index = 0; index < register_nb; index++)
+			read_data[index] = buffer[index + 1];
+	}
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Performs a burst read of FIFO buffer.
+ *
+ * @param dev        - The device structure.
+ * @param read_data  - The read values are stored in this buffer.
+ * @param bytes_nb	 - Number of bytes to be read in burst.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_get_fifo_value(struct adxl367_dev *dev,
```

Comment:
I would make this function static. If the user uses this function with an invalid bytes_nb he will receive invalid data.